metrics/sender.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package metrics
+
+import (
+	"time"
+
+	log "github.com/cihub/seelog"
+)
+
+// RoleProvider indicates that event was sent from service provider
+const RoleProvider = "provider"
+
+const appName = "myst"
+const startupEventName = "startup"
+
+// Sender builds events and sends them using given transport
+type Sender struct {
+	Transport Transport
+}
+
+// Transport allows sending events
+type Transport interface {
```

Comment:
Exported interface with all private methods - looks strange

Reply:
This allows passing around instancef of Transport, but hides internals - transports are meant only to be passed to `Sender` without exposing internals.
Later on, we can have `RetryTransport` which wraps another transport (i.e. HTTP!).